src/ui/ui_settings/ui_settings.js

```diff
@@ -0,0 +1,123 @@
+import { defaultsDeep } from 'lodash';
+import Bluebird from 'bluebird';
+
+import { getDefaultSettings } from './defaults';
+
+function hydrateUserSettings(user) {
+  return Object.keys(user).reduce(expand, {});
+  function expand(expanded, key) {
+    const userValue = user[key];
+    if (userValue !== null) {
+      expanded[key] = { userValue };
+    }
+    return expanded;
+  }
+}
+
+function assertRequest(req) {
+  if (
+    typeof req === 'object' &&
+    typeof req.path === 'string' &&
+    typeof req.headers === 'object'
+  ) return;
+
+  throw new TypeError('all uiSettings methods must be passed a hapi.Request object');
+}
+
+export class UiSettings {
+  constructor(server, status) {
+    this._server = server;
+    this._status = status;
+  }
+
+  getDefaults() {
```

Comment:
Shouldn't be a breaking change, there was one place where this method was used and I've updated it.